datafusion/physical-expr/src/datetime_expressions.rs

```diff
@@ -354,6 +354,24 @@ fn date_bin_impl(
                 }
             }
         }
+        ColumnarValue::Scalar(ScalarValue::IntervalMonthDayNano(Some(v))) => {
+            let (months, days, nanos) = IntervalMonthDayNanoType::to_parts(*v);
+            if months != 0 {
+                return Err(DataFusionError::NotImplemented(
+                    "DATE_BIN stride does not support month intervals".to_string(),
```

Comment:
👍  this is because months are not a fixed number of nanoseconds, right?

Reply:
Correct.

We'll need to round to the first of the month, and those nanosecond intervals are not regular. We'll probably be best served using the `chrono` package to handle month and year rounding.